docs/design_proposals/inspect-cmd.md

```diff
@@ -0,0 +1,65 @@
+# New command `skaffold inspect`
+
+* Author(s): gsquared94
+* Date: 2021-03-12
+* Status: [Under review]
+
+## Background
+
+Prior to the implementation of multiple configuration support in `skaffold` it was easy to take a look at the single `skaffold.yaml` file and understand the state of things. Now, a single `skaffold.yaml` configuration file can import other local and remote configurations as dependencies and we can have a complicated graph where it's not easy to predetermine what are all the artifacts that are going to be built and deployed by specifying a `module` filter when running skaffold (like `skaffold dev -m foo -m bar`).
+
+The Cloud Code IDEs (VSCode and IntelliJ) integrate with Skaffold and directly read and parse the `skaffold.yaml` file for information like the list of profiles available and the list of all artifacts.
+
+We should consolidate all such querying of the skaffold configuration behavior into a single command:
+```
+  skaffold inspect [flags] [options]
+```
+___
+
+## Design
+```
+skaffold inspect --help
+Display all resolved skaffold configurations.
+
+ You can use --output jsonpath={...} to extract specific values using a jsonpath expression. The configs defined in
+ the main `skaffold.yaml` file can be filted using the special `#` index; if there are multiple matches then it'll 
+ return a slice. 
+
+Examples:
+  # Show all resolved configurations with applied profiles.
+  skaffold inspect
+  
+  # Show all resolved configurations with profiles not applied.
+  skaffold inspect --raw
+  
+  # Get all named configs (modules) as a list
+  skaffold inspect -o jsonpath='{.configs[?(@.metadata.name != "")].metadata.name}'
+
+  # Get all profile names from the main `skaffold.yaml` configs.
+  skaffold inspect -o jsonpath='{.configs[#].profiles[*].name}'
```

Comment:
This seems powerful but complex — and is tied to the underlying representation; part of the motivation was to add some distance in having to know the details of the model and resolution.

Could we support the IDE cases explicitly with subcommands:
  - `skaffold inspect profiles`: list all profiles
  - `skaffold inspect artifacts`: list all artifacts
  -  `skaffold inspect configs`: list all config names

Should these list the underlying schema objects or an intermediate step?  (E.g., what should happen if we rework the skaffold model?)